test/plugin/base_test.rb

```diff
@@ -1623,14 +1676,25 @@ def setup_managed_vm_metadata_stubs
   def setup_container_metadata_stubs
```

Comment:
Might make sense to factor out `setup_common_container_metadata_stubs`, e.g.:
```ruby
  def setup_common_container_metadata_stubs(extra_attributes = [])
    attributes = [
      'attribute1', 'cluster-name', 'cluster-location', 'kube-env'
    ] + extra_attributes + ['last_attribute']
    stub_metadata_request('instance/attributes/', attributes.join("\n"))
    stub_metadata_request('instance/attributes/cluster-name',
                          CONTAINER_CLUSTER_NAME)
  end

  def setup_container_metadata_stubs
    setup_common_container_metadata_stubs
    stub_metadata_request('instance/attributes/kube-env',
                          "ENABLE_NODE_LOGGING: \"true\"\n"\
                          'INSTANCE_PREFIX: '\
                          "gke-#{CONTAINER_CLUSTER_NAME}-740fdafa\n"\
                          'KUBE_BEARER_TOKEN: AoQiMuwkNP2BMT0S')
  end

  def setup_container_metadata_stubs_kube_env_hidden
    setup_common_container_metadata_stubs
    stub_metadata_request_permission_denied('instance/attributes/kube-env')
  end

  def setup_cloudfunctions_metadata_stubs
    setup_common_container_metadata_stubs(['gcf_region'])
    stub_metadata_request('instance/attributes/kube-env',
                          "ENABLE_NODE_LOGGING: \"true\"\n"\
                          'INSTANCE_PREFIX: '\
                          "gke-#{CLOUDFUNCTIONS_CLUSTER_NAME}-740fdafa\n"\
                          "gke-#{CONTAINER_CLUSTER_NAME}-740fdafa\n"\
                          'KUBE_BEARER_TOKEN: AoQiMuwkNP2BMT0S')
    stub_metadata_request('instance/attributes/gcf_region',
                          CLOUDFUNCTIONS_REGION)
  end

  def setup_cloudfunctions_metadata_stubs_kube_env_hidden
    setup_common_container_metadata_stubs(['gcf_region'])
    stub_metadata_request_permission_denied('instance/attributes/kube-env')
    stub_metadata_request('instance/attributes/gcf_region',
                          CLOUDFUNCTIONS_REGION)
  end
```

Reply:
Given there's no longer any mention of `kube-env` in the entire plugin, can we just remove all the `kube-env` test stuff as well?

Reply:
We should ensure that the agent works in all environments where it previously worked. This includes those that supply kube-env. Let's keep those tests...

Reply:
Done, but why?  Given that `kube-env` isn't used anywhere in in the codebase, can't we safely assume that the behavior of that endpoint won't change the behavior of the plugin?